pkg/addon/addon_test.go

```diff
@@ -555,3 +558,76 @@ func TestGitLabReaderNotPanic(t *testing.T) {
 	_, err := NewAsyncReader("https://gitlab.com/test/catalog", "", "addons", "", gitType)
 	assert.EqualError(t, err, "git type repository only support github for now")
 }
+
+func TestCheckSemVer(t *testing.T) {
+	testCases := []struct {
+		actual   string
+		require  string
+		nilError bool
+		res      bool
+	}{
+		{
+			actual:  "v1.2.1",
+			require: "<=v1.2.1",
+			res:     true,
+		},
+		{
+			actual:  "v1.2.1",
+			require: ">v1.2.1",
+			res:     false,
+		},
+		{
+			actual:  "v1.2.1",
+			require: "<=v1.2.3",
+			res:     true,
+		},
+		{
+			actual:  "v1.2",
+			require: "<=v1.2.3",
+			res:     true,
+		},
+		{
+			actual:  "v1.2.1",
+			require: ">v1.2.3",
+			res:     false,
+		},
+		{
+			actual:  "v1.2.1",
+			require: "=v1.2.1",
+			res:     true,
+		},
+		{
+			actual:  "1.2.1",
+			require: "=v1.2.1",
+			res:     true,
+		},
+		{
+			actual:  "1.2.1",
+			require: "",
+			res:     true,
+		},
+	}
```

Comment:
please add more test cases about range:

* >v1.2.0 <=v1.2.3
* >v1.2.0 <v1.2.3
* >=v1.2.0 <v1.2.3
* >=v1.2.0 <=v1.2.3

Add number case:

* >v1.2.1 and  v1.2.10 should be true
* >v1.2.1 and v1.2.01 should be false

Reply:
why `v1.2.1 and v1.2.10 should be true`?

Reply:
added more tests